pkg/neonvm/controllers/vm_controller.go

```diff
@@ -311,6 +311,41 @@ func (r *VMReconciler) updateVMStatusMemory(
 	}
 }
 
+func (r *VMReconciler) acquireOverlayIP(ctx context.Context, vm *vmv1.VirtualMachine) error {
+	if vm.Spec.ExtraNetwork == nil || !vm.Spec.ExtraNetwork.Enable || len(vm.Status.ExtraNetIP) != 0 {
```

Comment:
IIUC, this is equivalent to: "if the extra network is disabled OR we already have an extra IP address, nothing to do" , right?

(worth adding that as a comment?)